### PR TITLE
chore: update table, list types

### PR DESCRIPTION
**Pull request by sendya**
# This is a ...

Update Table, List component typescript typo

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: types/list/list.d.ts [hunk 1]
```diff
@@ -3,104 +3,113 @@
...
+     * dataSource array for list
+     */
+    dataSource?: any[];
+
+    /**
+     * Custom item renderer, slot="renderItem" and slot-scope="item, index"
+     * @default null
+     * @type Function
+     */
+    renderItem?: (item: any, index?: number) => VNodeChild | JSX.Element;
```
Comment (tangjinzhou) [inline types/list/list.d.ts:106]: {item, index} 单参数

File: types/table/column.d.ts [hunk 2]
```diff
@@ -14,196 +13,205 @@ export interface ColumnFilterItem {
...
-    column: any,
-  ) => {
-    props: object;
-    attrs: object;
-    on: object;
-    class: object;
-    style: object;
-    nativeOn: object;
-  };
+export declare type CustomRenderFunction = (record: Record) => VNodeChild | JSX.Element;
```
Comment (tangjinzhou) [inline types/table/column.d.ts:22]: {text, record, index}

File: types/table/column.d.ts [hunk 3]
```diff
@@ -14,196 +13,205 @@ export interface ColumnFilterItem {
...
+    width?: string | number;
+
+    /**
+     * Set props on per cell
+     * @type Function
+     */
+    customCell?: (
+      record: any,
+      rowIndex: number,
+    ) => {
```
Comment (tangjinzhou) [inline types/table/column.d.ts:167]: 扁平化了  {class, style, ....}

File: types/table/column.d.ts [hunk 4]
```diff
@@ -14,196 +13,205 @@ export interface ColumnFilterItem {
...
+    };
+
+    /**
+     * Set props on per header cell
+     * @type object
+     */
+    customHeaderCell?: (
+      column: any,
+    ) => {
+      props: object;
```
Comment (tangjinzhou) [inline types/table/column.d.ts:183]: 扁平化了 {class, style, ....}

File: types/table/column.d.ts [hunk 5]
```diff
@@ -14,196 +13,205 @@ export interface ColumnFilterItem {
...
+     * @type object
+     */
+    slots?: object;
+
+    /**
+     * When using columns, you can use this property to configure the properties that support the slot-scope,
+     * such as scopedSlots: { customRender: 'XXX'}
+     * @type object
+     */
+    scopedSlots?: object;
```
Comment (tangjinzhou) [inline types/table/column.d.ts:215]: 去掉

File: types/table/table.d.ts [hunk 6]
```diff
@@ -119,186 +119,240 @@ export declare class Table extends AntdComponent {
...
+
+    /**
+     * Set props on per header row
+     * @type Function
+     */
+    customHeaderRow?: (
+      column: any,
+      index: number,
+    ) => {
+      props: object;
```
Comment (tangjinzhou) [inline types/table/table.d.ts:286]: 扁平化了 {class, style, ....}

File: types/table/table.d.ts [hunk 7]
```diff
@@ -119,186 +119,240 @@ export declare class Table extends AntdComponent {
...
+    /**
+     * Set props on per row
+     * @type Function
+     */
+    customRow?: (
+      record: any,
+      index: number,
+    ) => {
+      props: object;
+      attrs: object;
```
Comment (tangjinzhou) [inline types/table/table.d.ts:303]: 扁平化了 {class, style, ....}

File: types/table/column.d.ts (hunk 2 above)
Comment (sendya) [reply types/table/column.d.ts:22]:
```
export interface Record {
  text?: any;
  record?: object;
  index?: number;
}
```
这个已经调整了

**Pull request merged**